internal/clients/config/base.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type baseConfig struct {
+	Username  string
+	Password  string
+	ApiKey    string
+	UserAgent string
+	Header    http.Header
+}
+
+func newBaseConfigFromSDK(d *schema.ResourceData, version string, esKey string) baseConfig {
```

Comment:
Shall we rename it to something like `newESbaseConfigFromSDK`? The func is called only for `elasticsearch` attribute.

Reply:
Nope, as above it's not ES specific. It'd be inconsistent with everything else in this package IMO.